site/app/templates/course/CourseMaterials.twig

```diff
@@ -44,150 +45,129 @@
+                    <a class="key_to_click" tabindex="0" onclick='newDeleteCourseMaterialForm("{{ course_material.getPath | url_encode }}", "{{ name }}");' aria-label="Delete {{ name }}"> <i class="fas fa-trash" title="Delete the file" style="font-size: 16px; margin: 5px;"></i></a>
+                    <label for="date_to_release_{{ id }}">Share to student on</label>
+                    <input name="release_date" id="date_to_release_{{ id }}" class="date_picker" type="text" size="26"  value="{{ course_material.getReleaseDate|date(date_format) }}"/>
+                {% if not course_material.getSections|length == 0 %}
+                {% if course_material.isHiddenFromStudents %}
+                    <span>Sections (Hidden):
+                        {{course_material.getSections|map(s => s.getSectionId)|join(',')}}
                         </span>
-                    {% else %}
-                        <span>Sections:
-                        {{this_file_section|join(',')}}
+                {% else %}
+                    <span>Sections:
+                        {{course_material.getSections|map(s => s.getSectionId)|join(',')}}
                         </span>
-                    {% endif %}
-
                 {% endif %}
 
-                <script>
-                    function determineRelease(inputID){
-                        var now = new Date();
-
-                        function pad(str){
-                            return ('0'+str).slice(-2);
-                        }
-
-                        var date = now.getFullYear()+'-'+pad(now.getMonth()+1)+'-'+pad(now.getDate());
-                        var time = pad(now.getHours())+":"+pad(now.getMinutes())+":"+pad(now.getSeconds());
-                        var currentDT = date+' '+time;
-                        var neverDT = (now.getFullYear()+10)+'-'+pad(now.getMonth()+1)+'-'+pad(now.getDate())+' '+time;
-
-                        //get the value in each file so the color can be assigned
-                        //based on the time chosen
-                        var inputDT = document.getElementById(inputID);
-                        var fileDT = inputDT.value;
-
-                        fileDT = fileDT.replace(/\s/, 'T');
-                        currentDT = currentDT.replace(/\s/, 'T');
-                        neverDT = neverDT.replace(/\s/, 'T');
+                {% endif %}
 
-                        //also custom colors for this page for readability
-                        if(new Date(fileDT).getTime() <= new Date(currentDT).getTime()){
-                            return green;
-                        } else if(new Date(fileDT).getTime() >= new Date(neverDT).getTime()){
-                            return red;
-                        } else {
-                            return yellow;
+                    <script>
+                        function determineRelease(inputID){
+                            var now = new Date();
+
```

Comment:
Generally the use of `var` is discouraged. Perhaps we can change these twig functions to use `const` or `let` instead?